### PR TITLE
A Submittable attribute value can't be null

### PR DESCRIPTION
**Pull request by fpenim**
- Added AttributeValidator to check for null values;
- Remove unneeded `Jackson2JsonMessageConverter` message converter;
- Setup DLX for `core-validator`.

### CODE REVIEW

File: src/main/java/uk/ac/ebi/subs/validator/core/handlers/AssayHandler.java [hunk 1]
```diff
@@ -29,16 +34,17 @@
...
         SingleValidationResult singleValidationResult = new SingleValidationResult(ValidationAuthor.Core, assay.getId());
-
         studyRefValidator.validate(assay.getStudyRef(), singleValidationResult);
         sampleRefValidator.validateSampleUses(assay.getSampleUses(), singleValidationResult);
-
         checkValidationStatus(singleValidationResult);
+        resultList.add(singleValidationResult);
 
-        SingleValidationResultsEnvelope singleValidationResultsEnvelope = generateSingleValidationResultsEnvelope(envelope, Collections.singletonList(singleValidationResult));
+        resultList.addAll(attributeValidator.validate(assay.getAttributes(), assay.getId()));
```
Comment (ke4) [inline src/main/java/uk/ac/ebi/subs/validator/core/handlers/AssayHandler.java]: What will happen with the resultList? It is not used later. What is it purpose?
Probably the above mentioned refactor would solve this problem.

File: src/main/java/uk/ac/ebi/subs/validator/core/handlers/AssayDataHandler.java [hunk 2]
```diff
@@ -29,14 +33,17 @@
     @Override
     public SingleValidationResultsEnvelope handleValidationRequest(ValidationMessageEnvelope envelope) {
         AssayData assayData = (AssayData) envelope.getEntityToValidate();
+        List<SingleValidationResult> resultList = new ArrayList<>();
```
Comment (ke4) [inline src/main/java/uk/ac/ebi/subs/validator/core/handlers/AssayDataHandler.java]: I would refactor all the Handler class and move the `handleValidationRequest` method's duplicated code to the `AbstractHandler` class and have a submittable specific method in its subclasses. The `handleValidationRequest` method would call those methods.

File: src/main/java/uk/ac/ebi/subs/validator/core/validators/AttributeValidator.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    private static final String FAIL_MESSAGE = "An attribute value can't be null. Attribute - {}";
+
+    public List<SingleValidationResult> validate(List<Attribute> attributes, String id) {
+        List<SingleValidationResult> validationResults = new ArrayList<>();
+
+        if (attributes != null) {
+            attributes.forEach(attribute -> {
+                if (attribute.getValue() == null) {
+                    SingleValidationResult singleValidationResult = getDefaultSingleValidationResult(id);
+                    singleValidationResult.setMessage(String.format(FAIL_MESSAGE, attribute.getName()));
```
Comment (ke4) [inline src/main/java/uk/ac/ebi/subs/validator/core/validators/AttributeValidator.java]: Maybe it is an odd question: Can the name of the attribute be null?

File: build.gradle [hunk 4]
```diff
@@ -37,9 +37,10 @@ repositories {
 }
 
 dependencies {
-    compile ("uk.ac.ebi.subs:validator-common:0.3.0-SNAPSHOT")
+    compile ("uk.ac.ebi.subs:validator-common:0.4.0-SNAPSHOT")
```
Comment (ke4) [inline build.gradle]: If you do any refactor above, then you can change it to the latest: 0.5.0-SNAPSHOT (currently)

**Review by ke4 — DISMISSED** [review]
I added a small refactoring request.

**Review by davidrichardson — APPROVED** [review]

File: src/main/java/uk/ac/ebi/subs/validator/core/validators/AttributeValidator.java (hunk 3 above)
Comment (davidrichardson) [reply src/main/java/uk/ac/ebi/subs/validator/core/validators/AttributeValidator.java]: Agreed, that should not be permitted

**Review by davidrichardson — APPROVED** [review]

**Pull request merged**